editor/src/components/canvas/controls/size-box.tsx

```diff
@@ -134,15 +120,74 @@ interface ResizeEdgeProps {
   scale: number
   position: EdgePosition
   resizeStatus: ResizeStatus
+  targetComponentMetadata: ElementInstanceMetadata | null
+  labels: {
+    vertical: string
+    horizontal: string
+  }
+  dragState: DragState | null
+}
+
+interface ResizeEdgeState {
+  showLabel: boolean
+}
+
+function allowsInteractiveResize(layoutSystem: DetectedLayoutSystem): boolean {
+  switch (layoutSystem) {
+    case 'flex':
+    case 'flow':
+      return true
+    case 'grid':
+    case 'none':
+      return false
+    default:
+      const _exhaustiveCheck: never = layoutSystem
+      throw new Error(`Unhandled layoutSystem ${JSON.stringify(layoutSystem)}`)
+  }
+}
+
+function shiftPropertyTargetSelectorAxis(
+  primaryDirection: 'horizontal' | 'vertical',
+  actualDirection: 'horizontal' | 'vertical',
+  edge: 'before' | 'after',
+): number {
+  if (actualDirection === primaryDirection) {
+    if (edge === 'before') {
+      return 25
+    } else {
+      return 10
+    }
+  } else {
+    return 10
+  }
 }
 
-class ResizeEdge extends React.Component<ResizeEdgeProps> {
+class ResizeEdge extends React.Component<ResizeEdgeProps, ResizeEdgeState> {
```

Comment:
Was there a reason for using a class component here rather than a functional one?

Reply:
No idea, I assume it just hadn't been refactored as it's what is currently on master: https://github.com/concrete-utopia/utopia/blob/124cf4c96d226a559a68ce6ea4a5d8579794033b/editor/src/components/canvas/controls/size-box.tsx#L139

Reply:
Oh my bad, I read this as being completely new